v1/retryable_client.go

```diff
@@ -91,38 +91,34 @@ func (rc *RetryableClient) RetryableBulkImportRelationships(ctx context.Context,
 	})
 
 	_, err = bulkImportClient.CloseAndRecv() // transaction commit happens here
+	if err == nil {
+		return nil
+	}
 
 	// Failure to commit transaction means the stream is closed, so it can't be reused any further
 	// The retry will be done using WriteRelationships instead of BulkImportRelationships
 	// This lets us retry with TOUCH semantics in case of failure due to duplicates
 	retryable := isRetryableError(err)
 	conflict := isAlreadyExistsError(err)
 	canceled, cancelErr := isCanceledError(ctx.Err(), err)
-	unknown := !retryable && !conflict && !canceled && err != nil
+	unknown := !retryable && !conflict && !canceled
 
 	switch {
 	case canceled:
-
 		return cancelErr
 	case unknown:
-
 		return fmt.Errorf("error finalizing write of %d relationships: %w", len(relationships), err)
-	case conflict && conflictStrategy == Skip:
-
 	case retryable || (conflict && conflictStrategy == Touch):
 		err = rc.writeBatchesWithRetry(ctx, relationships)
 		if err != nil {
 			return fmt.Errorf("failed to write relationships after retry: %w", err)
 		}
+		return nil
 	case conflict && conflictStrategy == Fail:
 		return fmt.Errorf("duplicate relationships found")
-
 	default:
```

Comment:
as far as I can tell from the original code, the problem was introduced by returning an error in this `default` block. `unknown` is what captured any error that could have happened.

Now `unknown` variable does not capture any possible unknown error, hence this `default` block becomes necessary.

Wouldn't the `unknown` variable and case become unnecessary with the proposed code? `default` would capture `err != nil`.

Reply:
hey @vroldanbet - you are absolutely correct. The unknown check doesn't serve any purpose anymore. I've removed it